sycl/include/CL/sycl/detail/image_impl.hpp

```diff
@@ -239,12 +223,40 @@ class image_impl : public SYCLMemObjT {
              image_channel_type Type, const range<Dimensions> &ImageRange,
              const EnableIfPitchT<B> &Pitch, AllocatorT Allocator,
              const property_list &PropList = {})
-      : MAllocator(Allocator), MProps(PropList), MRange(ImageRange),
-        MOrder(Order), MType(Type),
-        MNumChannels(getImageNumberChannels(MOrder)),
+      : BaseT(PropList, Allocator), MRange(ImageRange), MOrder(Order),
+        MType(Type), MNumChannels(getImageNumberChannels(MOrder)),
         MElementSize(getImageElementSize(MNumChannels, MType)) {
     setPitches(Pitch);
-    handleHostData(HData);
+    BaseT::handleHostData(HData, detail::getNextPowerOfTwo(MElementSize));
+  }
+
+  image_impl(cl_mem MemObject, const context &SyclContext,
+             const size_t SizeInBytes, event AvailableEvent = {})
+      : BaseT(MemObject, SyclContext, std::move(AvailableEvent)) {
+    RT::PiMem Mem = pi::pi_cast<RT::PiMem>(BaseT::MInteropMemObject);
+    PI_CALL(RT::piMemGetInfo(Mem, CL_MEM_SIZE, sizeof(size_t), &BaseT::MSizeInBytes,
```

Comment:
&BaseT::MSizeInBytes is a pointer-to-member creation syntax and it is not what is expected here. Please change to &(BaseT::MSizeInBytes)